src/countries/countries.service.ts

```diff
@@ -1,4 +1,46 @@
 import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Country } from './country.entity';
+import { DeleteResult, Repository, UpdateResult } from 'typeorm';
+import { CountryInterface } from './country.interface';
 
 @Injectable()
-export class CountriesService {}
+export class CountriesService {
+  constructor(
+    @InjectRepository(Country)
+    private countryRepository: Repository<CountryInterface>,
```

Comment:
Always bind the repository to the entity not the Dto `private countryRepository: Repository<Country>`

Binding to the Dto will mismatch some database field since the Dto is only responsible for returning the fields that the user is interested in (ie. without the id), meanwhile the entity is meant to have the database representation as an object (ORM).

Currently, with the repository bind to the Dto, this will cause a TypeScript error:
```typescript
const country = await this.countryRepository.findOneBy({ id });
```

Since the Dto does NOT have an ID.